src/main/java/seedu/address/model/task/Type.java

```diff
@@ -30,10 +43,37 @@ public Type(String type) {
     }
 
     /**
-     * Returns true if a given string is a valid email.
+     * Returns true if a given string is a valid type.
+     */
+    public static boolean isValidType(String testType) {
+        if (testType == null) {
+            throw new NullPointerException("Type cannot be null");
+        }
+        for (TypeOfTask types : TypeOfTask.values()) {
+            String type = types.toString();
+            if (type.equals(testType)) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    /**
+     * Returns all possible types of task a user can create.
+     * @return a String containing all possible types of task a user can create.
      */
-    public static boolean isValidType(String test) {
-        return test.matches(VALIDATION_REGEX);
+    private static String listAcceptedTypes() {
+        StringBuilder sb = new StringBuilder();
+        int numberOfTypes = TypeOfTask.values().length;
+        int counter = 0;
+        for (TypeOfTask types : TypeOfTask.values()) {
+            sb.append(types.toString());
+            if (counter != numberOfTypes - 1) {
```

Comment:
Is there a reason to keep a counter? Don't we just need to traverse through all the types?

Reply:
this method is to list out the message constraints, in order to have a nice grammatically correct message the last type of task in this String should not contain a comma after it

e.g. `Acceptable types of tasks are: A, B, C.` instead of `Acceptable types of tasks are: A, B, C, .`

Reply:
hmmm maybe you can consider using the join method?

Reply:
Maybe you can consider using the "join" method?